src/cif.cc

```diff
@@ -0,0 +1,750 @@
+        for (int i = 0; i < n_models; ++i) {
+            auto chain_names = get_chains(models[i]);
+            int n_new_chains = chain_names->size();
+            n_chains += n_new_chains;
+
+            if (n_new_chains == 0) {
+                freesasa_warn("in %s(): no chains found (in model %s)",
+                              __func__, models[i].name.c_str());
+                continue;
+            }
+
+            ss.reserve(n_new_chains);
+            for (auto &chain_name : *chain_names) {
+                freesasa_structure *structure = structure_from_chain(doc, models[i].name, chain_name, classifier, options);
+                if (freesasa_structure_n(structure) == 0) {
+                    --n_chains;
+                    free(structure);
+                    continue;
+                }
+
+                ss.push_back(std::move(structure));
+
+                freesasa_structure_set_model(ss.back(), i + 1);
+            }
+        }
+        if (n_chains == 0)
+            freesasa_fail("In %s(): No chains in any model in protein: %s.",
+                          __func__, gemmi_struct.name.c_str());
+        *n = n_chains;
+    } else {
+        ss.reserve(n_models);
+        for (int i = 0; i < n_models; ++i) {
+            ss.emplace_back(
+                structure_from_model(doc, models[i].name, classifier, options));
+            freesasa_structure_set_model(ss.back(), i + 1);
+        }
+        *n = n_models;
+    }
+    return ss;
+}
+
+struct freesasa_MCRA {
+
+    freesasa_MCRA(const int model,
+                  const std::string &chain,
+                  const std::string &res_num,
+                  const std::string &residue,
+                  const std::string &atom)
+        : _model(model), _chain(chain), _res_num(res_num), _residue(residue), _atom(atom)
+    {
+    }
+
+    int find_row(gemmi::cif::Table &table, int start_idx = 0) const
+    {
+        int idx = start_idx, total_rows = table.length();
+
+        assert(idx < total_rows);
+
+        while (idx < total_rows) {
+            if (this->is_row(table[idx])) {
+                return idx;
+            }
+            ++idx;
+        }
+
+        freesasa_warn(
+            "In %s(), unable to find row in _atom_site for atom (%d, %s, %s, %s, %s). "
+            "Looping through entire table to double check...",
+            __func__, this->_model, this->_chain.c_str(),
+            this->_res_num.c_str(), this->_residue.c_str(),
+            this->_atom.c_str());
+
+        idx = 0;
+        for (const auto site : table) {
+            if (this->is_row(site)) {
+                return idx;
+            }
+            ++idx;
+        }
+        return FREESASA_FAIL;
+    }
+
+    friend std::ostream &operator<<(std::ostream &os, const freesasa_MCRA &mcra)
+    {
+        os << "Atom(" << mcra._model << " " << mcra._chain << " " << mcra._res_num << " [" << mcra._residue << "] " << mcra._atom << ")";
+        return os;
+    }
```

Comment:
I assume this was only used for debugging?

Reply:
I was initially for debugging but allows for a nice output of the atom. You can keep it or drop it. I don't think its currently being used anywhere.